python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -1843,6 +1843,58 @@ void basic_imported_symbols() {
     assertThat(((UnionType) mathModule.resolveMember("acos").get()).candidates()).allMatch(FunctionType.class::isInstance);
   }
 
+  @Test
+  void isInstanceTests() {
+    var xType = lastExpression("""
+      def foo():
+        x = 10
+        if isinstance(x, Foo):
+          ...
+        x
+      """).typeV2();
+
+    Assertions.assertThat(xType).isSameAs(PythonType.UNKNOWN);
```

Comment:
It's good to document the behavior anyway, but this is not what we'd expect (c.f my other comment).